viewing_party/party.py

```diff
@@ -1,23 +1,162 @@
+        
+    return top_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    user_movie_list = user_data["watched"]
+    friend_watched_list = user_data["friends"]
+    user_unique_movies = []
+    #------------------user data-------------------
+    for dictionary in user_movie_list:
+        if dictionary["title"] not in user_unique_movies:
+            user_unique_movies.append(dictionary)
+    
+    #--------------friend watched items-------------
+    for friends_dict in friend_watched_list:
+        for watched_list in friends_dict:
+            for dict in friends_dict[watched_list]:
+                for dictionary in user_unique_movies:
+                    if dictionary["title"] == dict["title"]:
+                        user_unique_movies.remove(dictionary)
+
+    return user_unique_movies
+
+def get_friends_unique_watched(user_data):
+    user_movie_list = user_data["watched"]
+    friend_watched_list = user_data["friends"]
+    friends_unique_movies = []
+
+    for friends_dict in friend_watched_list:
+        for dict in friends_dict["watched"]:
+            if dict["title"] == None:
+                return friends_unique_movies
+            if dict not in friends_unique_movies:
+                #### end up getting 4 instead of 3 movies in the final dictionary###
+                friends_unique_movies.append(dict)
+
+    for dictionary in user_movie_list:
+        for dict in friends_unique_movies:
+            if dictionary["title"] == dict["title"]:
+                #########does not remove common titles between user & friend. Look into it######
+                friends_unique_movies.remove(dict)
+
+    return friends_unique_movies
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    subscribed = user_data["subscriptions"]
+    not_seen_friends_watched_list = get_friends_unique_watched(user_data)
+    recommendations = []
+    for movie_dict in not_seen_friends_watched_list:
+        if movie_dict["host"] in subscribed:
+            recommendations.append(movie_dict)
+    
+    if recommendations == []:
+        return recommendations
+
+    return recommendations
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+    user_movie_list = user_data["watched"]
+    not_seen_friends_watched_list = get_friends_unique_watched(user_data)
+    users_top_genre = get_most_watched_genre(user_data)
+    recommendations = []
+
+    for movie in not_seen_friends_watched_list:
+        if movie["genre"] == users_top_genre and movie not in user_movie_list:
+            recommendations.append(movie)
+    if recommendations == []:
+        return recommendations

```

Comment:
You don't need this guard clause here because the recommendations start out empty.